app/plugin_api_test.go

```diff
@@ -556,6 +556,36 @@ func TestPluginAPIUserCustomStatus(t *testing.T) {
 	userCs, err = th.App.GetCustomStatus(user1.Id)
 	assert.Nil(t, err)
 	assert.Equal(t, csClear, userCs)
+
+	// Set a custom status for clearing it later
+	custom.Emoji = ":100"
+	custom.Text = "happy 100"
+
+	err = api.UpdateUserCustomStatus(user1.Id, custom)
+	assert.Nil(t, err)
+	userCs, err = th.App.GetCustomStatus(user1.Id)
+	assert.Nil(t, err)
+	assert.Equal(t, custom, userCs)
+
+	// Let plugin set a status
+	pluginCs := &model.CustomStatus{
+		Emoji:        ":calendar:",
+		Text:         "In meeting",
+		SetByProduct: true,
+	}
+	err = api.UpdateUserCustomStatus(user1.Id, pluginCs)
+	assert.Nil(t, err)
+	userCs, err = th.App.GetCustomStatus(user1.Id)
```

Comment:
I think `custom` is the "current" status right? Then `userCs` is the user-provided one, and `pluginCs` is the plugin-provided one

```suggestion
	custom, err = th.App.GetCustomStatus(user1.Id)
```

Reply:
`userCs` is just the custom status we get back everytime we call `GetCustomStatus`. It is used to check against the expect custom status.